label_studio_ml/examples/openai/openai_predictor.py

```diff
@@ -0,0 +1,84 @@
+import pickle
+import os
+import requests
+import json
+from uuid import uuid4
+
+from label_studio_ml.model import LabelStudioMLBase
+from label_studio_ml.utils import DATA_UNDEFINED_NAME, get_env
+import openai
+import re
+
+
+HOSTNAME = get_env('HOSTNAME', 'http://localhost:8080')
+API_KEY = get_env('API_KEY')
+openai.api_key = os.environ['OPENAI_API_KEY']
+
+print('=> LABEL STUDIO HOSTNAME = ', HOSTNAME)
+if not API_KEY:
+    print('=> WARNING! API_KEY is not set')
+
+
+class OpenAIPrecitor(LabelStudioMLBase):
+
+    def __init__(self, **kwargs):
+        # don't forget to initialize base class...
+        super(OpenAIPrecitor, self).__init__(**kwargs)
+
+        # Parsed label config contains only one output of <Choices> type
+        assert len(self.parsed_label_config) == 1
+        self.from_name, self.info = list(self.parsed_label_config.items())[0]
+        assert self.info['type'] == 'Choices'
+
+        # the model has only one textual input
+        assert len(self.info['to_name']) == 1
+        assert len(self.info['inputs']) == 1
+        assert self.info['inputs'][0]['type'] == 'Text'
+        self.to_name = self.info['to_name'][0]
+        self.value = self.info['inputs'][0]['value']
+
+    def _get_sentiment(self, input_text):
+        prompt = f"Respond in the json format: {{'response': sentiment_classification}}\nText: {input_text}\nSentiment (Positive, Neutral, Negative):"
```

Comment:
alternatively - these prompts, output classes and other GPT parameters can be passed via
```
label-studio-ml start gpt --with prompt="..." model="gpt-3.5-turbo" 
```

then retrieve from `__init__(**kwargs)`

Reply:
Pulling out the model definitely makes sense. I think the prompt might be more difficult, since in this case you want the `input text` to be mid prompt. But you would need to change it whenever your classes change. It probably depens on how closely we want to tie the prompt and the config.